### PR TITLE
feat: Click to select annotations

### LINKED ISSUE #84: Click to select annotations
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Add the ability to select paintbrush or spline annotations with a single click on the annotation. When selected, an annotation becomes active and can then be modified or deleted.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by SilviaZeta**
# Description

Add the ability to select paintbrush or spline annotations with a single click on the annotation. When selected, an annotation becomes active and can then be modified or deleted.

For paintbrush annotations you can toggle between draw and select mode with ctrl+a (to try this out, draw multiple annotations, ctrl+a, click on existing annotations and delete, ctrl+a and draw some more..).

closes #84 

# Dependency changes

No

# Testing

No

# Documentation

N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ChasNelson1990 — CHANGES_REQUESTED** [review]
Can we use a different cursor for select mode? Maybe the pointy finger hand?

**Comment by ChasNelson1990** [comment]
Also, do we want to enforce the same keyboard shortcut for both paintbrushes and splines just for consistency?

File: src/toolboxes/spline/Canvas.tsx [hunk 1]
```diff
@@ -325,10 +327,82 @@ export class SplineCanvas extends Component<Props, State> {
...
+    const dx = point.x - point1.x;
+    const dy = point.y - point1.y;
+    const dxLine = point2.x - point1.x;
+    const dyLine = point2.y - point1.y;
+    // Use the cross-product to check whether the XYpoint lies on the line passing
+    // through XYpoint 1 and XYpoint 2.
+    const crossProduct = dx * dyLine - dy * dxLine;
+    // If the XYpoint is exactly on the line the cross-product is zero. Here we set a threshold
+    // based on ease of use, to accept points that are close enough to the spline.
+    if (Math.abs(crossProduct) > 700) return false;
```
Comment (cooper667) [inline src/toolboxes/spline/Canvas.tsx:381]: Maybe this value should be a constant at the top?

**Review by cooper667 — APPROVED** [review]

**Comment by ChasNelson1990** [comment]
Also, one more change to propose, can we make the active paintbrush a standard colour so its clear that you've activated the right one? Or give it a stroke*? We could be the active paintbrush palette:primary? as I think the colour when your painting is palette:secondary?
* if we go for stroke let's make it that pink (palette: secondary?)

**Comment by SilviaZeta** [comment]
> Also, one more change to propose, can we make the active paintbrush a standard colour so its clear that you've activated the right one? Or give it a stroke*? We could be the active paintbrush palette:primary? as I think the colour when your painting is palette:secondary?
> 
> * if we go for stroke let's make it that pink (palette: secondary?)

Can we open a new issue for this?

File: src/toolboxes/paintbrush/Canvas.tsx [hunk 2]
```diff
@@ -54,12 +58,15 @@ export class PaintbrushCanvasClass extends Component<Props, State> {
 
   private points: XYPoint[];
 
+  private mode: number;
```
[the remark this answers is not in the document]
Comment (SilviaZeta) [reply src/toolboxes/paintbrush/Canvas.tsx]: Typescript is happy with number.

**Comment by cooper667** [comment]
> Also, one more change to propose, can we make the active paintbrush a standard colour so its clear that you've activated the right one? Or give it a stroke*? We could be the active paintbrush palette:primary? as I think the colour when your painting is palette:secondary?
> 
> * if we go for stroke let's make it that pink (palette: secondary?)

I think this will be solved when we do opacity on the strokes? Solid for selected, 50% opaque for others?

File: src/toolboxes/paintbrush/Canvas.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SilviaZeta) [reply src/toolboxes/paintbrush/Canvas.tsx]: Typescript is also happy with Mode, so I've changed this as Mode is more specific.

[the remark this answers is not in the document]
Comment (cooper667) [reply src/toolboxes/paintbrush/Canvas.tsx]: 👍  thanks

**Review by ChasNelson1990 — APPROVED** [review]

**Review by cooper667 — APPROVED** [review]

**Pull request merged**